erpnext/patches/v8_1/set_delivery_date_in_so_item.py

```diff
@@ -8,14 +8,14 @@ def execute():
 		frappe.db.sql("""
 			update `tabSales Order`
 			set delivery_date = final_delivery_date
-			where (delivery_date is null or delivery_date = '' or delivery_date = '0000-00-00')
+			where (delivery_date is null or delivery_date is null or delivery_date = '0000-00-00')
 				and order_type = 'Sales'""")
 
 	frappe.db.sql("""
 		update `tabSales Order` so, `tabSales Order Item` so_item
 		set so_item.delivery_date = so.delivery_date
 		where so.name = so_item.parent
 			and so.order_type = 'Sales'
-			and (so_item.delivery_date is null or so_item.delivery_date = ''
+			and (so_item.delivery_date is null or so_item.delivery_date is null
```

Comment:
I don't think `so_item.delivery_date = ''` is causing the error. Is the patch run after this fix?

I think the fix should be adding another condition like below:
```
and (so.delivery_date is not null and so.delivery_date != '' and so.delivery_date != '0000-00-00')
```